Applications/Spire/Source/Ui/TextBox.cpp

```diff
@@ -301,6 +422,16 @@ void TextBox::update_display_text() {
   }
 }
 
+void TextBox::update_placeholder_text() {
+  if(is_placeholder_shown()) {
+    m_placeholder->setText(get_elided_text(m_placeholder->fontMetrics(),
```

Comment:
Although the font metrics have been obtained from m_placeholder, the elided placeholder text is still calculated by m_line_edit in the get_elided_text function rather than m_placeholder.
auto option = QStyleOptionFrame();
  option.initFrom(m_line_edit);
  option.rect = m_line_edit->contentsRect();
 ...
  auto rect = m_line_edit->style()->subElementRect(QStyle::SE_LineEditContents,
    &option, m_line_edit);

The placement text area is computed by m_line_edit.

 I am worried that it could cause some potential issues. For example, after you click the Reset button and then copy a long text which can make it overflow to the placeholder (not input a text), the placeholder text can’t be elided.

Reply:
The elision issue has been fixed.